src/protocol/commands/_qXfer_exec_file.rs

```diff
@@ -0,0 +1,34 @@
+use super::prelude::*;
+
+use crate::common::Pid;
+
+#[derive(Debug)]
+pub struct qXferExecFileRead<'a> {
+    pub pid: Option<Pid>,
+    pub offset: usize,
+    pub length: usize,
+
+    pub buf: &'a mut [u8],
+}
+
+impl<'a> ParseCommand<'a> for qXferExecFileRead<'a> {
+    fn from_packet(buf: PacketBuf<'a>) -> Option<Self> {
+        let (buf, body_range) = buf.into_raw_buf();
+        let body = buf.get_mut(body_range.start..body_range.end)?;
+
+        if body.is_empty() {
+            return None;
+        }
+
+        let mut body = body.split(|b| *b == b':').skip(1);
+        let pid = decode_hex(body.next()?).ok().and_then(Pid::new);
```

Comment:
This implementation isn't correct, in two respects:

1. it fails if the client leaves the `pid` annex empty. 
2. it doesn't propagate errors from `decode_hex` correctly

wrt. 1: In the current implementation, `decode_hex` would be passed an empty slice, thereby returning a `DecodeHexError::Empty`, which would result in a packet parse error. Instead, we need to manually check if `body.next()?` is an empty slice, and set the `pid = None` ourselves.

wrt 2: if `decode_hex` returns an error, the `.ok()` will return `None`, but instead of propogating that error via `?`, it is chained to `and_then(Pid::new)`, which will silently return `None` as well, resulting in the packet "successfully parsing" with `pid = None`. This is not correct, and we should be early-returning if the packet fails to parse.

A correct implementation therefore would therefore be as follows:

```suggestion
        let pid = match body.next()? {
            [] => None,
            buf => Some(Pid::new(decode_hex(buf).ok()?)?)
        }
```